### DISCUSSION TITLE
GitHub Repository Name vs Description vs README Heading H1

### DISCUSSION THREAD

**Discussion by jpkmw** [community/community#23793] [New to GitHub]
<p>I am looking for a  guide or a naming convention documentation that explains best practices for your GitHub Repository Name vs the Description Field vs the H1 Heading of the README.</p>
<p>I am working on a Naming Convention Guide for my company and the situation I want to use in the guide is where the Repository Name will have 1 name that is short and separated by dashes “-”</p>
<p>The Description will be a longer more descriptive version of the Repository Name, and the Heading H1 will usually be somewhere in between in terms of Length of title. </p>
<p>I am looking for documentation that explains the relationship between the 3 GitHub sections and how to go about naming each section. </p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi <a class="mention" href="/jpkmw">@jpkmw</a>! :wave:</p>
<p>There isn’t really one true naming convention - different people come up with different conventions to suit themselves.</p>
<p>Having a short repository name that is alphanumeric strings separated by dashes <code>-</code> makes sense! You could also potentially use periods <code>.</code>. Dashes and periods are the only punctuation marks allowed in repo names.</p>
<p>One thing to consider with the description field is that it can be searched:</p>
<p><a href="https://help.github.com/en/github/searching-for-information-on-github/searching-for-repositories#search-by-repository-name-description-or-contents-of-the-readme-file" class="onebox" target="_blank">https://help.github.com/en/github/searching-for-information-on-github/searching-for-repositories#search-by-repository-name-description-or-contents-of-the-readme-file</a></p>
<p>…so you might want to put key words in there!</p>
<p>And then the H1 of the README is going to be the prominent title that people see when they land on the repo. It doesn’t necessarily need to repeat the repository name. It could be a call to action, a title to establish context, or anything else you think might be useful.</p>
